### PR TITLE
new packages: biopieces and dependencies

### PR DESCRIPTION
**Pull request by akthoma**
Adding new package biopieces and its perl/ruby gem dependencies. I also edited the ruby recipe to comply with the AutotoolsPackage format. 

I know this is an overwhelming amount of additions. I did not separate into separate branches to prevent a package being merged without its dependency, as the dependency tree for biopieces is very large.

### CODE REVIEW

File: var/spack/repos/builtin/packages/biopieces/package.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
```
Comment (junghans) [inline var/spack/repos/builtin/packages/biopieces/package.py]: Change to spack/spack!

File: var/spack/repos/builtin/packages/perl-try-tiny/package.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
```
Comment (junghans) [inline var/spack/repos/builtin/packages/perl-try-tiny/package.py]: Same

File: var/spack/repos/builtin/packages/ruby-gnuplot/package.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
```
Comment (junghans) [inline var/spack/repos/builtin/packages/ruby-gnuplot/package.py]: Same

File: var/spack/repos/builtin/packages/scan-for-matches/package.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    """scan_for_matches is a utility written in C for locating patterns in DNA
+       or protein FASTA files."""
+
+    homepage = "http://blog.theseed.org/servers/2010/07/scan-for-matches.html"
+    url      = "http://www.theseed.org/servers/downloads/scan_for_matches.tgz"
+
+    version('2010-7-16', 'f64c9cfb385984ded2a7ad9ad2253d83')
+
+    def install(self, spec, prefix):
+        cc = which('cc')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/scan-for-matches/package.py]: Use `self.compiler.cc`

**Comment by akthoma** [comment]
@junghans how do you recommend I split the branch into multiple PRs? They all depend on each other. I don't want something getting merged without its dependency.

**Comment by akthoma** [comment]
is this good to go?

**Comment by akthoma** [comment]
Bump, please 😄  I leave my organization on Wednesday so I would prefer to have this either merged by then or be aware of what I need to fix (if a fix is required).

File: var/spack/repos/builtin/packages/biopieces/package.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    depends_on('perl-dbfile', type=('build', 'run'))
+    depends_on('perl-dbd-mysql', type=('build', 'run'))
+
+    depends_on('ruby@1.9:')
+    depends_on('ruby-gnuplot')
+    depends_on('ruby-narray')
+    depends_on('ruby-rubyinline')
+    depends_on('ruby-terminal-table')
+
+    depends_on('python@2:6', type=('build', 'run'))
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/biopieces/package.py]: @akthoma 

This means "a version of python between 2 and 6 (inclusive)". Did you mean that this should only run with python `2.6`? If so, at the moment Spack's python package doesn't keep track of a `2.6` version.

Other than that this all looks good to me

Comment (scheibelp) [reply var/spack/repos/builtin/packages/biopieces/package.py]: In other words if I pull this branch and try `spack spec biopieces` then I get `python@2.7.14` as a dependency. If that is expected, I think a constraint like `depends_on('python@2:', type=...)` makes more sense.

Comment (akthoma) [reply var/spack/repos/builtin/packages/biopieces/package.py]: Whoops! I meant `@2.6:`. Thanks!

File: var/spack/repos/builtin/packages/biopieces/package.py [hunk 6]
```diff
@@ -64,7 +64,7 @@ class Biopieces(Package):
     depends_on('ruby-rubyinline')
     depends_on('ruby-terminal-table')
 
-    depends_on('python@2:6', type=('build', 'run'))
+    depends_on('python@2.6;', type=('build', 'run'))
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/biopieces/package.py]: I think you'd want that to be a `:` vs. a `;`

Comment (akthoma) [reply var/spack/repos/builtin/packages/biopieces/package.py]: you're right. I'm struggling today

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**